src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -138,13 +154,66 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
     this._selectInitialTrackIfNeeded();
   }
 
+  /**
+   * Check that a period has a supported track for the given track type.
+   * Trigger an error with `MANIFEST_INCOMPATIBLE_CODECS_ERROR` if no codecs are
+   * supported and `onTracksNotPlayableForType[ttype]` is "error".
+   *
+   * @param period - The period to check
+   * @param ttype - The track type.
+   */
+  private checkPeriodHasSupportedTrack(
+    period: IPeriodMetadata,
+    ttype: "video" | "audio",
+  ): void {
+    const adaptationsForType = period.adaptations[ttype];
+    const periodHasAdaptationForType =
+      adaptationsForType !== undefined && adaptationsForType.length > 0;
+    if (!periodHasAdaptationForType) {
+      // The content does not have audio/video media, e.g. it's an audio only content.
+      // there is no track of this type to select.
+      log.debug(
+        `TS: The period does not have adaptation for ${ttype} there is no track to choose`,
+      );
+      return;
+    }
+
+    if (
+      adaptationsForType.every((adapt) => adapt.supportStatus.hasSupportedCodec === false)
+    ) {
+      const firstPlayableAdaptation = findFirstPlayableAdaptation(period, ttype);
+      const err = new MediaError(
+        "MANIFEST_INCOMPATIBLE_CODECS_ERROR",
+        "No supported " + ttype + " adaptations",
+        { tracks: undefined },
+      );
+
+      if (
```

Comment:
It seems that in this particular case, we do not have the `noPlayableTrack` event.

Is that wanted? From reading the API documentation, I would think that this case would lead to that event.

Reply:
Though after doing more thinking, it seems that sending the `noPlayableTrack` event here might not work because we might not have advertised the Period yet through a `newAvailablePeriods` event, no?